java-extensions/jdbc-bridge/src/main/java/com/starrocks/jdbcbridge/DataSourceCache.java

```diff
@@ -0,0 +1,27 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+package com.starrocks.jdbcbridge;
+
+import com.zaxxer.hikari.HikariDataSource;
+
+import java.util.HashMap;
+import java.util.Map;
+
+// a cache for get datasource
+public class DataSourceCache {
+    private final Map<String, HikariDataSource> sources = new HashMap<>();
+    private static final DataSourceCache instance = new DataSourceCache();
+
+    public static DataSourceCache getInstance() {
+        return instance;
+    }
+
+    public synchronized HikariDataSource getSource(String driver, String user, String passwd, String url) {
```

Comment:
do we need locks for getSource and put? the synchronized keyword can't guarantee that getSource and put are atomic, right? maybe there are two threads calling getSource and get null, and then creating a new DataSource at the same time

Reply:
keep getSource and put atomic is enough.
it's safe to create two datasource and we can only cache one datasource

Reply:
OK I use putIfAbsent instead of it